docs/src/transaction/procedures.md

```diff
@@ -0,0 +1,122 @@
+# Kernel Procedures
+
+The transaction kernel provides a set of procedures that can be invoked by account code, note scripts, and transaction scripts. These procedures are organized into several categories and have limitations on which context they can be called from, which are documented here.
+
+## Contexts
+
+This describes contexts from which procedures can be called:
+
+- **Any**: Can be called from any context.
+- **Account**: Can only be called from native or foreign accounts.
+- **Native**: Can only be called when the current account is the native account.
+- **Auth**: Can only be called from the authentication procedure. Since it is called on the native account, it implies **Native** and **Account**.
+- **Note**: Can only be called from a note script.
+- **Faucet**: Can only be called when the current account is a faucet.
+
+If a procedure has multiple context requirements they are combined using `&`. For instance, "Native & Account" means the procedure can only be called when the current account is the native one _and_ only from the account context.
+
+## Procedures
```

Comment:
Reading through these, I'm wondering if maybe describing the "internal" procedures is a bit too low level. These procedures will never be called directly - so, the docs are basically for kernel developers (i.e., us).

Maybe it would be better to describe the "wrapper" procedures from the `miden` namespace. These almost always map one-to-one to the underlying kernel procedures, but are the procedures users might use. So, instead of `account_get_initial_commitment` we'd have `get_initial_commitment` in the `miden::account` module etc.